google/cloud/bigtable/backup.py

```diff
@@ -0,0 +1,332 @@
+    def start_time(self):
+        """The time this Backup was started.
+
+        :rtype: :class:`datetime.datetime`
+        :returns: A 'datetime' object representing the time when the creation
+                  of this Backup had started.
+        """
+        return self._start_time
+
+    @property
+    def end_time(self):
+        """The time this Backup was finished.
+
+        :rtype: :class:`datetime.datetime`
+        :returns: A 'datetime' object representing the time when the creation
+                  of this Backup was finished.
+        """
+        return self._end_time
+
+    @property
+    def size_bytes(self):
+        """The size of this Backup, in bytes.
+
+        :rtype: int
+        :returns: The size of this Backup, in bytes.
+        """
+        return self._size_bytes
+
+    @property
+    def state(self):
+        """ The current state of this Backup.
+
+        :rtype: :class:`~google.cloud.bigtable_admin_v2.gapic.enums.Backup.State`
+        :returns: The current state of this Backup.
+        """
+        return self._state
+
+    @classmethod
+    def from_pb(cls, backup_pb, instance):
+        """Creates a Backup instance from a protobuf message.
+
+        :type backup_pb: :class:`table_pb2.Backup`
+        :param backup_pb: A Backup protobuf object.
+
+        :type instance: :class:`Instance <google.cloud.bigtable.instance.Instance>`
+        :param instance: The Instance that owns the Backup.
+
+        :rtype: :class:`~google.cloud.bigtable.backup.Backup`
+        :returns: The backup parsed from the protobuf response.
+        :raises: ValueError: If the backup name does not match the expected
+                             format or the parsed project ID does not match the
+                             project ID on the Instance's client, or if the
+                             parsed instance ID does not match the Instance ID.
+        """
+        match = re.compile(
+            r"^projects/(?P<project>[^/]+)/"
+            r"instances/(?P<instance_id>[a-z][-a-z0-9]*)/"
+            r"clusters/(?P<cluster_id>[a-z][-a-z0-9]*)/"
+            r"backups/(?P<backup_id>[a-z][a-z0-9_\-]*[a-z0-9])$"
+        ).match(backup_pb.name)
+        if match is None:
+            raise ValueError(
+                "Backup protobuf name was not in the expected format.", backup_pb.name
+            )
+        if match.group("project") != instance._client.project:
+            raise ValueError(
+                "Project ID of the Backup does not match the Project ID "
+                "of the instance's client"
+            )
+
+        instance_id = match.group("instance_id")
+        if instance_id != instance.instance_id:
+            raise ValueError(
+                "Instance ID of the Backup does not match the Instance ID "
+                "of the instance"
+            )
+        backup_id = match.group("backup_id")
+        return cls(backup_id, instance)
+
+    def create(self, cluster_id=None):
```

Comment:
should we also optionally ask for an `expire_time` and `table_id` as well, to give user an option to pass all the `create` required arguments as params?

Also probably need a `@table_id.setter` similar to `@cluster.setter` and `@expire_time.setter` to give user an option to set this parameter since it is optional in the backup constructor.

Reply:
:+1: Totally correct. Added all the fields found in `backup_pb2`.